firebase-auth/src/androidMain/kotlin/dev/gitlive/firebase/auth/auth.kt

```diff
@@ -113,9 +113,28 @@ public actual class FirebaseAuth internal constructor(public val android: com.go
     public actual fun useEmulator(host: String, port: Int): Unit = android.useEmulator(host, port)
 }
 
-public actual class AuthResult internal constructor(public val android: com.google.firebase.auth.AuthResult) {
+public actual class AuthResult(
```

Comment:
why cant the constructor stay internal?

Reply:
I just feel like there is no need for it to be internal. Having it public just allows more flexibility for the library. People can define their own extensions for stuff that is platform specific etc (like my example) and still keep the common types from the library.

I mean sure, in my example I could redefine the `AuthResult` in my own code and use that instead and this PR kinda removes the need for that code all together. But still the constructor being internal seems like an unnecessary obstacle forcing me to write something that is already there, while offering not much benefit elsewhere.

It's definitely up to discussion, depends if there are any more serious reasons for it to be internal in the first place.